crates/tabby/src/services/completion.rs

```diff
@@ -321,35 +322,51 @@ impl CompletionService {
             self.config.max_decoding_tokens,
         );
 
+        let mut use_crlf = false;
         let (prompt, segments, snippets) = if let Some(prompt) = request.raw_prompt() {
             (prompt, None, vec![])
         } else if let Some(segments) = request.segments.as_ref() {
+            let mut new_segments = segments.clone();
+            if segments.prefix.contains("\r\n") {
+                use_crlf = true;
+                new_segments.prefix = segments.prefix.replace("\r\n", "\n");
```

Comment:
this replacement only applies for prefix / suffix, what about other segments? I suggest do deteciton on prefix / suffix, but the replacement shall happens on the whole prompt / output.